library/std/src/sys/pal/windows/mod.rs

```diff
@@ -60,7 +60,7 @@ pub unsafe fn init(_argc: isize, _argv: *const *const u8, _sigpipe: u8) {
 
     // Normally, `thread::spawn` will call `Thread::set_name` but since this thread already
     // exists, we have to call it ourselves.
-    thread::Thread::set_name(&c"main");
+    thread::Thread::set_name_wide(&['m' as u16, 'a' as u16, 'i' as u16, 'n' as u16, 0]);
```

Comment:
It bugs me that `b"main\0".map(|b| b as u16)` works but `c"main".map(|b| b as u16)` doesn't.

Reply:
Hm, I should probably make a macro to do const encoding of `&str` to null terminated `&[u16]`.

Reply:
I went ahead and made a little macro for const utf16 strings. I just did a pretty direct translation of the wikipedia articles.